### ISSUE TITLE
Bug Report when deploying root module for creating management groups and archetypes

### ISSUE THREAD

**Issue by SirSteel11** [Azure/terraform-azurerm-caf-enterprise-scale#476]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 0.15.1 and 1.0.6

**azure provider**: 3.0.2

**module**: caf-enterprise-scale (management groups and policy)

### Description 
I am trying to deploy the core structure trough an Azure devops pipeline. When I cone to the "plan" phase of deployment I get an unexpected error 

#### Describe the bug
│ Error: Unsupported attribute
│ 
│   on .terraform\modules\enterprise_scale\locals.management_groups.tf line 292, in locals:
│  292:         access_control = value.archetype_config.access_control
│     ├────────────────
│     │ value.archetype_config is object with 2 attributes
│ 
│ This object does not have an attribute named "access_control".
╵
##[error]Error: Terraform Plan failed with exit code: 1

#### Steps to Reproduce
Tried the default deployment as listed in this repo
https://github.com/Azure/terraform-azurerm-caf-enterprise-scale#maintf

#### Screenshots
![image](https://user-images.githubusercontent.com/89302262/193405368-f54b243c-72ce-4596-8119-10872118843b.png)

**Comment by jtracey93** [comment]
Hey @SirSteel11,

I have just tested the default example that you linked to in Cloud Shell and it planned successfully as below:
![image](https://user-images.githubusercontent.com/41163455/193399894-2de48a98-36dd-4788-b65b-ba4a03668e25.png)

Have you customized any part of the variables and inputs? If so, can you provide them (with sensitive info redacted) here please or via a [gist](https://gist.github.com/)?

Also, what version of terraform are you using? you can find out by running `terraform version` in the pipeline as a step 👍

Let us know

Thanks

Jack

**Comment by jtracey93** [comment]
Sorry have seen your TF versions were provided (it's early here 😆) 

This may be related to #353 but this has been resolved from what I can see and recall.

Looping in @matt-FFFFFF & @krowlandson also here as they were involved on that issue also.

Id strongly suggest checking and custom archetypes in a lib are correctly formatted and casing is all lowercase as per defaults: https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/main/modules/archetypes/lib/archetype_definitions/archetype_definition_default_empty.json

**Comment by SirSteel11** [comment]
@jtracey93 Thanks for a swift reply!
To try and correct the error I cloned this .git and added   
library_path  = "${path.root}/lib" to my configuration.
I then copied the cloned "lib" -folder to the root of mine .git repo. I then created a new statefile key and rerunned the pipeline again with the same outcome as the initial error. I have added my configuration files in this comment 
[core-governance-dynamic.zip](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/files/9690041/core-governance-dynamic.zip)
The task is running on a computer node with following profile:
Windows server 2016
Agent version 2.181.2
